services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers_tasks.py

```diff
@@ -0,0 +1,317 @@
+import logging
+from textwrap import dedent
+from typing import Optional
+
+from fastapi import APIRouter, Depends, FastAPI, HTTPException, Request, status
+from servicelib.fastapi.long_running_tasks.server import (
+    TaskAlreadyRunningError,
+    TaskId,
+    TaskManager,
+    get_task_manager,
+    start_task,
+)
+from servicelib.fastapi.requests_decorators import cancel_on_disconnect
+
+from ..core.rabbitmq import RabbitMQ
+from ..core.settings import ApplicationSettings
+from ..models.schemas.application_health import ApplicationHealth
+from ..models.schemas.containers import ContainersCreate
+from ..models.shared_store import SharedStore
+from ..modules.long_running_tasks import (
+    task_containers_restart,
+    task_create_service_containers,
+    task_ports_inputs_pull,
+    task_ports_outputs_pull,
+    task_ports_outputs_push,
+    task_restore_state,
+    task_runs_docker_compose_down,
+    task_save_state,
+)
+from ..modules.mounted_fs import MountedVolumes
+from ._dependencies import (
+    get_application,
+    get_application_health,
+    get_mounted_volumes,
+    get_rabbitmq,
+    get_settings,
+    get_shared_store,
+)
+
+logger = logging.getLogger(__name__)
+
+containers_router_tasks = APIRouter(tags=["containers"])
+
+
+# HANDLERS
+
+
+@containers_router_tasks.post(
+    "/containers",
+    summary=dedent(
+        """
+    Starts a background task responsible for:
```

Comment:
suggestion:
```
Starts the containers as defined in ContainerCreate by:
  - cleaning up resources from previous runs if any
  - pulling the needed images
  - starting the containers

  Progress may be obtained throuhg URL
  Process may be cancelled through URL
```

Side question: what happens if I call this API although I already called it before? what do I get?

Reply:
Ok I get the 409. Maybe that is the only error I would document. 422 is something I would anyway expect.

Reply:
> Side question: what happens if I call this API although I already called it before? what do I get?

You get a 409, another task is already being handled.